proofs/eo/cpc/programs/AciNorm.eo

```diff
@@ -13,8 +13,8 @@
 ; note: >
 ;   This method is similar in purpose to $singleton_elim, but insists that
 ;   the function and its nil terminator are provided explicitly.
-(program $singleton_elim_aci ((T Type) (S Type) (U Type) (W Type) (f (-> T U S)) (id S) (x S) (x1 T) (x2 U :list))
-  ((-> T U S) W W) W
+(program $singleton_elim_aci ((S Type) (U Type) (f (-> S S S)) (id U) (x U) (x1 U) (x2 U))
+  ((-> S S S) U U) U
```

Comment:
There are typing issues here... why does `f` take `x1` and `x2` of type `U` instead of `S`? Similarly for the rest of the file...

Reply:
For example, `f` could be `bvor` which is `(-> (BitVec m) (BitVec m) (BitVec m))`, whereas the other arguments can be `(BitVec 4)`, so `S = (BitVec m), U = (BitVec 4)`.  Generally `U` is expected to be an instance of `S`.